packages/frontend/components/Header.tsx

```diff
@@ -1,14 +1,40 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 import { TAutoConnect, useAutoConnect } from "~~/hooks/scaffold-eth";
 import { ConnectButton } from "@rainbow-me/rainbowkit";
 import { Faucet } from "~~/components/scaffold-eth";
+import { useBalance } from "wagmi";
 
 // todo: move this later scaffold config.  See TAutoConnect for comments on each prop
 const tempAutoConnectConfig: TAutoConnect = {
   enableBurnerWallet: true,
   autoConnect: true,
 };
 
+const HeaderBalance = ({ address, balanceSymbol }: { address: string; balanceSymbol: string }) => {
+  const [balance, setBalance] = useState<number | null>(null);
+  const {
+    data: fetchedBalanceData,
+    // isError,
+    // isLoading,
+  } = useBalance({
+    addressOrName: address,
+    watch: true,
+    cacheTime: 30_000,
+  });
+
+  useEffect(() => {
+    if (fetchedBalanceData?.formatted) {
+      setBalance(Number(fetchedBalanceData.formatted));
+    }
+  }, [fetchedBalanceData]);
+
+  return (
+    <span className="m-2">
+      {balance?.toFixed(2)} {balanceSymbol}
+    </span>
+  );
+};
+
```

Comment:
Instead of creating this, lets use the component that we already have (`Balance`).